.github/workflows/main.yml

```diff
@@ -13,22 +13,22 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: 16
-      - name: Get npm cache directory
-        id: npm-cache
+      - name: Get yarn cache directory
+        id: yarn-cache
         run: |
-          echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
+          echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
       - uses: actions/cache@v3
         with:
-          path: ${{ steps.npm-cache.outputs.dir }}
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
-            ${{ runner.os }}-node-
+            ${{ runner.os }}-yarn-
```

Comment:
I have since learned (and setup-node developed), that it's better (and much easier to maintain) to simply use the caching option from setup-node.

```
uses: actions/setup-node
with:
  cache: 'yarn'
```